pointcloud/point.go

```diff
@@ -126,11 +126,39 @@ func (bp *basicData) Value() int {
 	return bp.value
 }
 
-func (bp *basicData) SetIntensity(v uint16) Data {
-	bp.intensity = v
-	return bp
-}
+// MarshalBinary checks for the presence of color and value data. If present, it will store said data in an order byte
+// array.
+func (bp *basicData) MarshalBinary() (data []byte, err error) {
+	var dataBytes []byte
+
+	if bp.HasColor() {
+		c := color.NRGBAModel.Convert(bp.Color()).(color.NRGBA)
 
-func (bp *basicData) Intensity() uint16 {
-	return bp.intensity
+		dataBytes = append(dataBytes, c.R)
+		dataBytes = append(dataBytes, c.G)
+		dataBytes = append(dataBytes, c.B)
+		dataBytes = append(dataBytes, c.A)
+	}
+	if bp.HasValue() {
+		dataBytes = append(dataBytes, byte(bp.Value()))
+	}
+	return dataBytes, nil
+}
+
+// UnmarshalBinary takes in a bytes array and checking the length and setting the color and values appropriately.
+func (bp *basicData) UnmarshalBinary(dataBytes []byte) error {
+	switch len(dataBytes) {
```

Comment:
If `value` becomes a variable number of bytes, or increases, then this switch statement would need to change too.